spec/order_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'order'
+require 'menu'
+
+describe Order do
+  subject(:main_order) { described_class.new }
+
+  let(:order) do
+    {
+      hotdog: 2,
+      fries: 1
+    }
+  end
+
+  it "responds to add a dish" do
+    expect(main_order).to respond_to(:add).with(2).argument
+  end
+
+  it "selects several dishes from the menu" do
+    main_order.add(:hotdog, 2)
+    main_order.add(:fries, 1)
+    expect(main_order.order).to eq(order)
+  end
+
+  it "tells if a dish is on the menu" do
+    expect(main_order.has_dish?(:hotdog)).to be true
+  end
+
+  it "tells if a dish is not on the menu" do
+    expect(main_order.has_dish?(:hamburger)).to be false
+  end
+
+  it "doesn't allow items to be added that are not on the menu" do
+    allow(main_order).to receive(:has_dish?).with(:hotdog).and_return(true)
```

Comment:
When we do mocking (doubles and stubs) in a unit test for a class, we want to mock _other_ classes and the methods contained in those _other_ classes. We don't want to mock the class we are actually unit testing, because then we aren't testing the real functionality of the class we are aiming to unit test.